gammapy/datasets/flux_points.py

```diff
@@ -275,107 +296,102 @@ def plot_residuals(self, ax=None, method="diff", **kwargs):
 
         Parameters
         ----------
-        ax : `~matplotlib.pyplot.Axes`
-            Axes object.
-        method : {"diff", "diff/model", "diff/sqrt(model)"}
-            Method used to compute the residuals, see `MapDataset.residuals()`
+        ax : `~matplotlib.axes.Axes`
+            Axes to plot on.
+        method : {"diff", "diff/model"}
+            Normalization used to compute the residuals, see `FluxPointsDataset.residuals`.
         **kwargs : dict
-            Keyword arguments passed to `~matplotlib.pyplot.errorbar`.
+            Keyword arguments passed to `~matplotlib.axes.Axes.errorbar`.
 
         Returns
         -------
-        ax : `~matplotlib.pyplot.Axes`
+        ax : `~matplotlib.axes.Axes`
             Axes object.
         """
         import matplotlib.pyplot as plt
 
-        ax = plt.gca() if ax is None else ax
-
-        residuals = self.residuals(method=method)
+        ax = ax or plt.gca()
 
         fp = self.data
+        residuals = self.residuals(method)
 
         xerr = fp._plot_get_energy_err()
         if xerr is not None:
             xerr = xerr[0].to_value(self._e_unit), xerr[1].to_value(self._e_unit)
 
-        model = self.flux_pred()
         yerr = fp._plot_get_flux_err(fp.sed_type)
-
         if method == "diff":
             unit = yerr[0].unit
             yerr = yerr[0].to_value(unit), yerr[1].to_value(unit)
         elif method == "diff/model":
+            model = self.flux_pred()
             unit = ""
-            yerr = (yerr[0] / model).to_value(""), (yerr[1] / model).to_value(unit)
+            yerr = (yerr[0] / model).to_value(unit), (yerr[1] / model).to_value(unit)
         else:
-            raise ValueError("Invalid method, choose between 'diff' and 'diff/model'")
+            raise ValueError('Invalid method, choose between "diff" and "diff/model"')
 
+        kwargs.setdefault("color", kwargs.pop("c", "black"))
         kwargs.setdefault("marker", "+")
-        kwargs.setdefault("ls", "None")
-        kwargs.setdefault("color", "black")
+        kwargs.setdefault("linestyle", kwargs.pop("ls", "none"))
 
-        ax.errorbar(
-            self.data.e_ref.value, residuals.value, xerr=xerr, yerr=yerr, **kwargs
-        )
+        ax.errorbar(fp.e_ref.value, residuals.value, xerr=xerr, yerr=yerr, **kwargs)
+        ax.axhline(0, color=kwargs["color"], lw=0.5)
 
         # format axes
-        ax.axhline(0, color="black", lw=0.5)
-        ax.set_ylabel("Residuals {}".format(unit.__str__()))
-        ax.set_xlabel(f"Energy ({self._e_unit})")
+        ax.set_xlabel(f"Energy [{self._e_unit}]")
```

Comment:
Which energy is it? reconstructed or true? It might be useful that this is always precised in the plot function, for non-TeV experts...

Reply:
Of course it's not the case in general, but for the flux points computed with Gammapy it is reconstructed energy. At some point we introduced the convention, that "energy" refers to reconstructed energy, while we make it explicit for true energy (https://docs.gammapy.org/dev/irf/index.html#irf-axis-naming). So the axis label is fine here I think...